core/drivers/crypto/caam/hash/local.h

```diff
@@ -0,0 +1,133 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright 2019-2020 NXP
+ *
+ * CAAM hash/HMAC local header.
+ */
+#ifndef __LOCAL_H__
+#define __LOCAL_H__
+
+#include <caam_common.h>
+
+/*
+ * Full hashing/hmac data SW context
+ */
+struct hashctx {
+	uint32_t *descriptor;	   /* Job descriptor */
+	struct caamblock blockbuf; /* Temporary block buffer */
+	struct caambuf ctx;	   /* Hash context used by the CAAM */
+	const struct hashalg *alg; /* Reference to the algo constants */
+	struct caambuf key;	   /* HMAC split key */
+};
+
+/*
+ * Hash/hmac Algorithm definition
+ */
+struct hashalg {
+	uint32_t type;	     /* Algo type for operation */
+	uint8_t size_digest; /* Digest size */
+	uint8_t size_block;  /* Computing block size */
+	uint8_t size_ctx;    /* CAAM context register size (8 + digest size) */
+	uint8_t size_key;    /* HMAC split key size */
+};
+
+/* First part CAAM HW context - message length */
+#define HASH_MSG_LEN 8
+
+/*
+ * Constants definition of the hash/hmac algorithm
+ */
+static const struct hashalg hash_alg[] = {
```

Comment:
This will be duplicated in each C file including this file. Please find a way to avoid that.

Reply:
You're right, please check latest commit.